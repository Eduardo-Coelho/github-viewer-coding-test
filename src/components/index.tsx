export * from './commitDetails/CommitDetails';
export * from './commitTable/CommitTable';
export * from './repoSearch/RepoSearch';
export * from './loader/Loader';
export * from './modal/Modal';
