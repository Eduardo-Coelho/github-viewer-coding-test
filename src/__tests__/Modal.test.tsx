import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Modal } from '../components';

test('Modal matches snapshot when open', () => {
  const { asFragment } = render(
    <Modal isOpen={true} onClose={() => {}}>
      <div>Modal Content</div>
    </Modal>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Modal matches snapshot when closed', () => {
  const { asFragment } = render(
    <Modal isOpen={false} onClose={() => {}}>
      <div>Modal Content</div>
    </Modal>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Modal calls onClose when Escape key is pressed', () => {
  const handleClose = jest.fn();
  render(
    <Modal isOpen={true} onClose={handleClose}>
      <div>Modal Content</div>
    </Modal>,
  );

  fireEvent.keyDown(document, { key: 'Escape', code: 'Escape' });
  expect(handleClose).toHaveBeenCalledWith(true);
});

test('Modal calls onClose when Close button is clicked', () => {
  const handleClose = jest.fn();
  const { getByText } = render(
    <Modal isOpen={true} onClose={handleClose}>
      <div>Modal Content</div>
    </Modal>,
  );

  fireEvent.click(getByText('Close'));
  expect(handleClose).toHaveBeenCalledWith(true);
});
