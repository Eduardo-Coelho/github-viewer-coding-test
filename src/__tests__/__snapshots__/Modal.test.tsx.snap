// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot when closed 1`] = `<DocumentFragment />`;

exports[`Modal matches snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="model-container"
  >
    <div
      class="model-content"
    >
      <div>
        Modal Content
      </div>
      <button
        class="model-button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
